### PR TITLE
Allocate all emitter var field strings statically

### PR DESCRIPTION
**Pull request by gzm0**
Memory profile analysis shows ~120k duplicates of the string "$n" amounting to ~6MB waste (on the test suite).

Further digging revealed that this happens through the "$n" call-helper (null check) via:

- SJSGen#genCallHelper
- VarGen#globalVar
- VarGen#globalVarIdent
- VarGen#genericIdent

By defining the relevant strings statically (including the leading `$`), we avoid the duplication.

Discovered while working on https://github.com/scala-js/scala-js/issues/4906.

### CODE REVIEW

**Comment by gzm0** [comment]
Based on #4909 due to merge conflicts.

**Review by sjrd — CHANGES_REQUESTED** [review]
Very nice. I only have a few comments, most of them typos.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarFld.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+package org.scalajs.linker.backend.emitter
+
+/** Namespace for generated fields.
+ *
+ *  Mainly to avoid duplicate strings in memory.
+ *
+ *  Also gives us additional compile-time safety against typos.
+ */
+private[emitter] final class VarFld private (val str: String) extends AnyVal
```
[the remark this answers is not in the document]
Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/VarFld.scala:21]: Only brevity. And it's true that saving 2 chars is likely not worth the additional abbreviation.

**Comment by gzm0** [comment]
I have addressed all the comments. I have also added `private[emitter]` to `object VarField`.

**Review by sjrd — APPROVED** [review]

**Pull request merged**